nautobot/dcim/forms.py

```diff
@@ -688,6 +688,29 @@ class Meta:
             "outer_unit": StaticSelect2(),
         }
 
+    def clean(self):
+        cleaned_data = self.cleaned_data
+        site = cleaned_data.get("site")
+
+        if self.instance:
+            # If the site is changed, the rack post save signal attempts to update the rack devices,
+            # which may result in an Exception.
+            # To avoid an unhandled exception in signal, catch this error here.
+            duplicate_devices_names = (
+                Device.objects.values_list("name", flat=True)
+                .annotate(name_count=Count("name"))
+                .filter(name_count__gt=1)
+            )
+            if duplicate_devices := (
+                Device.objects.filter(site=site, name__in=list(duplicate_devices_names))
+                .exclude(rack=self.instance)
+                .values_list("name", flat=True)
+            ):
```

Comment:
It would not work since the rack's device has not yet been changed to the new 'site,' resulting in it being filtered out even if it has a duplicate name.

Reply:
Ah okay, then is the `.exclude(rack=self.instance)` unnecessary in that case?